source/common/http/header_map_impl.h

```diff
@@ -37,6 +37,14 @@ public:
  */
 class HeaderMapImpl : public HeaderMap {
 public:
+  /**
+   * Appends data to header. If header already has a value, the string ',' is added between the
```

Comment:
Q: Is this RFC particular about whether it should be ',' or ", "? I feel like in practice there is a space after the comma, but I'm not sure. (I think this is what we do for appending XFF FWIW, and on that topic can we have XFF append use this new code?).

Reply:
I believe whitespace is optional.  We do direct comma appending internally except for the Via header which we special case ", " with no explanation of why.  Via needs whitespace between received-protocol and received-by but as documented it appears to be optional between the , and received-protocol.
https://tools.ietf.org/html/rfc7230#page-82     OWS (optional whitespace)

Reply:
Also https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html
(which I'm failing to find subsection links for)

Multiple message-header fields with the same field-name MAY be present in a message if and only if the entire field-value for that header field is defined as a comma-separated list [i.e., #(values)]. It MUST be possible to combine the multiple header fields into one "field-name: field-value" pair, without changing the semantics of the message, by appending each subsequent field-value to the first, each separated by a comma. The order in which header fields with the same field-name are received is therefore significant to the interpretation of the combined field value, and thus a proxy MUST NOT change the order of these field values when a message is forwarded.

Reply:
OK cool I'm fine either way. It would be good to be consistent everywhere we can which I guess this will do for us.

Reply:
Worth linking to RFC here for whitespace? I'm fine either way.